### DISCUSSION TITLE
How to perform third-party JS validation when saving entries in CP

### DISCUSSION THREAD

**Discussion by michaelrog** [craftcms/cms#10754] [Q&A] [answered]
I'm looking for docs for hooking into an "event" in the CP, before an entry is saved, to run some custom logic, and possibly interrupt/cancel the save action.

(The logic I have in mind isn't related to a field that I control, so I can't accomplish this via field UI as far as I know.)

I suppose I could just listen for a `click` event on the Save button itself, but that seems fragile: I don't know how the markup for the save button might change in future Craft versions, and I also don't want to be competing with other event listeners pr request lifecycles that Craft has attached.

What's the best way to do this?

**Comment by brandonkelly** [comment, marked as answer]
There’s no provided framework for this. I think your best bet would be to hook into the draft validation, and add validation errors from there.

**Reply by michaelrog** [reply]
My use case is confirmation, not validation. So I need to ask for the user's input _before_ submitting the form, and the actual field content isn't relevant to the outcome...
